modules/ztp-create-factory-hub-cluster.adoc

```diff
@@ -0,0 +1,58 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/ztp-factory-install-clusters.adoc
+:_content-type: CONCEPT
+[id="preparing-the-factory-install-environment_{context}"]
+= Preparing the factory install environment
+
+.Prerequisites
+Before you can provision edge clusters at scale, you must meet the following prerequisites:
+
+=== Base
+
+* Deploy the {product-title} cluster with three control plane nodes following the guidance in the section "Deploying installer-provisioned clusters on bare metal". Alternatively you can use the technology preview Assisted Installer from link:https://cloud.redhat.com/[cloud.redhat.com] to create the cluster. To install single node OpenShift follow the guidance in "Installing on a single node" in the "Installing" section of the {product-title} documentation.
+** All Cluster Operators are available.
+** Cluster is reachable using a `KUBECONFIG` file.
+** The API, API-INT and ingress should be deployed on edge cluster on the DHCP external network.
+
+=== Storage
+
+* Storage can be provided by installing the Local Storage Operator and by using local volumes or by using OpenShift Data Foundation (ODF). Deploy ODF following the guidance in link:https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.9/html/deploying_openshift_data_foundation_using_bare_metal_infrastructure/index[Deploying OpenShift Data Foundation using bare metal infrastructure].
+
+[NOTE]
+====
+If the cluster is greater than 3 nodes, the recommendation is to use OpenShift Data Foundation. If it is a single-node OpenShift cluster, use the Local Storage Operator.
+====
+
+** Create the following persistent volumes with at least 200GB of storage (NVMe or SSD) for:
+
+    *** 2 for Assisted Installer.
+    *** 1 for the hub internal registry that is for the mirror of the images. At least 200GB is required on the hub, more may be required if ODF is installed.
+    *** 1 for HTTPD that hosts the Red Hat Enterprise Linux CoreOS (RHCOS) images.
```

Comment:
This assembly is designated for OKD too, but there are some places that might not be appropriate for OKD, like this line. (You should be able to use the {op-system-first} attribute so that it switches to FCOS properly for OKD). 

But I'd recommend to do a look at the OKD version with an SME/QE too to make sure that there's nothing else that is incorrect for OKD. For example, somewhere it's replaced as the "OKD Pipelines Operator", which I'm guessing isn't quite right?

Reply:
I reached out to manager running the project and this has not been tested on OKD distro. To be honest I notice OpenShift Pipelines are not I OKD docs. I said I would remove from OKD and he agreed with that approach.